app/user-setup/index.tsx

```diff
@@ -0,0 +1,215 @@
+                    className="m-2 rounded border px-4 py-2"
+                    onPress={openImageLibrary}
+                >
+                    <Text>Upload new image</Text>
+                </TouchableOpacity>
+                <TouchableOpacity
+                    className="m-2 rounded border px-4 py-2"
+                    onPress={openCamera}
+                >
+                    <Text>Take a picture</Text>
+                </TouchableOpacity>
+            </View>
+            <View className="mt-5">
+                <Text className="my-2 text-xl font-extrabold">
+                    Display name
+                </Text>
+                <Controller
+                    control={control}
+                    name="name"
+                    render={({ field: { onChange, value } }) => (
+                        <TextInput
+                            className="border-hairline h-14 rounded-md border-black px-3 py-1.5 text-lg text-gray-900"
+                            placeholder="Enter Display Name"
+                            onChangeText={onChange}
+                            value={value}
+                        />
+                    )}
+                />
+                {errors.name && (
+                    <Text className="mt-1 text-red-500">
+                        {errors.name.message}
+                    </Text>
+                )}
+            </View>
+            <View className="flex-col">
+                <View className="mb-2 mt-4 flex-row justify-start">
+                    <Text className="text-xl font-extrabold">Campus </Text>
+                    <Text className="text-xl color-gray-400">
+                        (preferred location)
+                    </Text>
+                </View>
+                {CampusEnum.options.map((campus, _) => {
+                    return (
+                        <TouchableOpacity
+                            key={campus}
+                            className={clsx(
+                                "my-1 grow justify-end rounded-lg px-2 py-3",
+                                campus === campusSelected
+                                    ? "border border-blue-500"
+                                    : "border-hairline border-dashed bg-white",
+                            )}
+                            onPress={() => handleSetCampus(campus)}
+                        >
+                            <Text
+                                className={clsx(
+                                    "text-center text-lg",
+                                    campus === campusSelected
+                                        ? "font-extrabold color-blue-400"
+                                        : "font-semibold color-black",
+                                )}
+                            >
+                                {campus}
+                            </Text>
+                        </TouchableOpacity>
+                    );
+                })}
```

Comment:
extract to another component maybe named `CampusSelectionButtons`, store in `/components/user/campus-buttons.tsx`

just pass in the necessary props

- use Button from ui-kitten
- make the buttons disabled when `isLoading` == true